File: src/main/kotlin/com/learning/gpt/config/GptConfig.java
package com.learning.gpt.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestTemplate;

import static org.springframework.http.MediaType.APPLICATION_JSON;

@Configuration
public class GptConfig {

    @Value("${openai.api-key}")
    private String apiKey;

    @Bean
    public RestTemplate restTemplate(){
        RestTemplate template = new RestTemplate();
        template.getInterceptors().add((request, body, execution) -> {
            request.getHeaders().add(
                    "Authorization"
                    ,"Bearer " + apiKey);
            return execution.execute(request, body);
        });

        return template;
    }
}

File: src/main/kotlin/com/learn/ai/config/GptConfig.kt
package com.learn.ai.config

import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestTemplate

@Configuration
@EnableConfigurationProperties(value = [GptProperties::class])
class GptConfig(
    val gptProperties: GptProperties,
) {

        @Bean
        fun restTemplate(): RestTemplate {
            return RestTemplate()
        }
}